Warning for potentialCollisionPoints in Windows builds

@sherm1 brought this to my attention. In windows builds we get the warning:

```
DrakeCollision.cpp

..\..\drake/systems/plants/collision/bullet_model.h(120): warning C4373: 'DrakeCollision::BulletModel::potentialCollisionPoints': virtual function overrides 'DrakeCollision::Model::potentialCollisionPoints', previous versions of the compiler did not override when parameters only differed by const/volatile qualifiers

c:\projects\workspace\experimental\a99ae8ad\drake\systems\plants\collision\Model.h(145): note: see declaration of 'DrakeCollision::Model::potentialCollisionPoints'
```

See for instance [here](https://drake-cdash.csail.mit.edu/viewBuildError.php?type=1&buildid=42360).

--- a/drake/systems/plants/collision/DrakeCollision.cpp
+++ b/drake/systems/plants/collision/DrakeCollision.cpp
@@ -42,7 +42,7 @@ unique_ptr<Model> newModel() {
 #ifdef BULLET_COLLISION
   return newModel(BULLET);
 #else
-  DRAKE_ABORT_UNLESS("DrakeCollision must be compiled with Bullet.");
+  DRAKE_ABORT_UNLESS(!"DrakeCollision must be compiled with Bullet.");
 #endif
 }
 };

--- a/drake/systems/plants/collision/DrakeCollision.cpp
+++ b/drake/systems/plants/collision/DrakeCollision.cpp
@@ -21,7 +21,7 @@ enum DRAKECOLLISION_EXPORT ModelType { NONE, AUTO, BULLET };
 unique_ptr<Model> newModel(ModelType model_type) {
   switch (model_type) {
     case NONE:
-      return unique_ptr<Model>(new Model());
+      return nullptr;
       break;
     case BULLET:
 #ifdef BULLET_COLLISION
@@ -42,7 +42,7 @@ unique_ptr<Model> newModel() {
 #ifdef BULLET_COLLISION
   return newModel(BULLET);
 #else
-  return newModel(NONE);
+  DRAKE_ABORT_UNLESS("DrakeCollision must be compiled with Bullet.");
 #endif
 }
 };

--- a/drake/systems/plants/collision/Model.h
+++ b/drake/systems/plants/collision/Model.h
@@ -142,8 +142,7 @@ class DRAKECOLLISION_EXPORT Model {
    * \return a vector of PointPair objects containing the potential
    * collision points
    */
-  virtual std::vector<PointPair> potentialCollisionPoints(
-      const bool use_margins) {
+  virtual std::vector<PointPair> potentialCollisionPoints(bool use_margins) {
     return std::vector<PointPair>();
   }
 

--- a/drake/systems/plants/collision/Model.h
+++ b/drake/systems/plants/collision/Model.h
@@ -42,7 +42,7 @@ class DRAKECOLLISION_EXPORT Model {
   /** \brief Perform any operations needed to bring the model up-to-date
    * after making changes to its collision elements
    */
-  virtual void updateModel() {}
+  virtual void updateModel() = 0;
 
   /** \brief Change the element-to-world transform of a specified collision
    * element.
@@ -66,9 +66,8 @@ class DRAKECOLLISION_EXPORT Model {
    */
   virtual bool closestPointsAllToAll(const std::vector<ElementId>& ids_to_check,
                                      const bool use_margins,
-                                     std::vector<PointPair>& closest_points) {
-    return false;
-  }
+                                     std::vector<PointPair>&
+                                     closest_points) = 0;
 
   /** Computes the point of closest approach between collision elements that
    are in contact.
@@ -82,9 +81,7 @@ class DRAKECOLLISION_EXPORT Model {
    @returns `true` if this method ran successfully and `false` otherwise.
    **/
   virtual bool ComputeMaximumDepthCollisionPoints(
-      const bool use_margins, std::vector<PointPair>& closest_points) {
-    return false;
-  }
+      const bool use_margins, std::vector<PointPair>& closest_points) = 0;
 
   /** \brief Compute the points of closest approach between specified pairs
    * of collision elements
@@ -99,9 +96,8 @@ class DRAKECOLLISION_EXPORT Model {
    */
   virtual bool closestPointsPairwise(const std::vector<ElementIdPair>& id_pairs,
                                      const bool use_margins,
-                                     std::vector<PointPair>& closest_points) {
-    return false;
-  }
+                                     std::vector<PointPair>&
+                                     closest_points) = 0;
 
   /** Clears possibly cached results so that a fresh computation can be
   performed.
@@ -117,7 +113,7 @@ class DRAKECOLLISION_EXPORT Model {
   computation without any coupling with previous history.
 
   @see drake/systems/plants/collision/test/model_test.cc. **/
-  virtual void ClearCachedResults(bool use_margins) {}
+  virtual void ClearCachedResults(bool use_margins) = 0;
 
   /** \brief Compute closest distance from each point to any surface in the
    * collision model utilizing Bullet's collision detection code.
@@ -129,7 +125,7 @@ class DRAKECOLLISION_EXPORT Model {
    */
   virtual void collisionDetectFromPoints(
       const Eigen::Matrix3Xd& points, bool use_margins,
-      std::vector<PointPair>& closest_points) {}
+      std::vector<PointPair>& closest_points) = 0;
 
   /** \brief Compute the set of potential collision points for all
    * eligible pairs of collision geometries in this model. This includes
@@ -142,9 +138,7 @@ class DRAKECOLLISION_EXPORT Model {
    * \return a vector of PointPair objects containing the potential
    * collision points
    */
-  virtual std::vector<PointPair> potentialCollisionPoints(bool use_margins) {
-    return std::vector<PointPair>();
-  }
+  virtual std::vector<PointPair> potentialCollisionPoints(bool use_margins) = 0;
 
   /** Given a vector of points in world coordinates, returns the indices
   of those points within a specified distance of any collision geometry in
@@ -168,9 +162,7 @@ class DRAKECOLLISION_EXPORT Model {
   @see systems/plants/test/collidingPointsTest.m for a Matlab test. **/
   virtual std::vector<size_t> collidingPoints(
       const std::vector<Eigen::Vector3d>& input_points,
-      double collision_threshold) {
-    return std::vector<size_t>();
-  }
+      double collision_threshold) = 0;
 
   /** Tests if any of the points supplied in input_points collides with
   any part of the model within a given threshold.
@@ -191,9 +183,7 @@ class DRAKECOLLISION_EXPORT Model {
   `false` otherwise. **/
   virtual bool collidingPointsCheckOnly(
       const std::vector<Eigen::Vector3d>& input_points,
-      double collision_threshold) {
-    return false;
-  }
+      double collision_threshold) = 0;
 
   /** Performs raycasting collision detecting (like a LIDAR / laser rangefinder)
    *
@@ -210,9 +200,7 @@ class DRAKECOLLISION_EXPORT Model {
   virtual bool collisionRaycast(const Eigen::Matrix3Xd& origin,
                                 const Eigen::Matrix3Xd& ray_endpoint,
                                 bool use_margins, Eigen::VectorXd& distances,
-                                Eigen::Matrix3Xd& normals) {
-    return false;
-  }
+                                Eigen::Matrix3Xd& normals) = 0;
 
   /**
    * Modifies a collision element's local transform to be relative to a joint's

